cranelift-sizeless-vector.md

```diff
@@ -39,8 +107,6 @@ This doesn't mean that a backend can't select a fixed width during code generati
 # Open questions
 [open-questions]: #open-questions
 
-- Does anyone care that Cranelift could only support a maximum of 128 vector lanes? The only problem I could imagine is if someone has a 128-lane vector of 1-bit bools...
-- How will the register allocator (regalloc2..?) handle a new vector type and/or potential register aliasing?
-- Are there parts of Cranelift, which aren't backend specific, that would need to handle these types? (is there generic stack handling or anything else data size specific...?)
+- How will regalloc2 handle a new vector type and/or potential register aliasing? And will sizeless spill slots be possible?
```

Comment:
We can definitely talk about designs that allow for different spill slot sizes; I can imagine a notion of "multiple spill slot spaces" (i.e., different index domains managed separately).

But I think the simplest approach here is what we had talked about before: allocate vector registers as the basic unit; they are as wide as they are, and both v128-typed values and bigger-sizeless-typed values allocate one vector register. The spillslot then needs to be as big as the actual register width but otherwise this scheme seems reasonable enough to me.

(FWIW, regalloc.rs and regalloc2 have the same design regarding spillslot abstractions and API, so the same considerations apply both before and after migration)